WcaOnRails/app/controllers/posts_controller.rb

```diff
@@ -73,7 +74,10 @@ def destroy
     #  | 2014 |
     #  +------+
     #  1 row in set, 1 warning (0.00 sec)
-    world_readable_posts = Post.where(world_readable: true)
-    world_readable_posts.find_by_slug(params[:id]) || world_readable_posts.find_by_id!(params[:id])
+    post = Post.find_by_slug(params[:id]) || Post.find_by_id(params[:id])
+    if !post || !post.world_readable
+      raise ActiveRecord::RecordNotFound.new("Couldn't find post")
+    end
+    post
```

Comment:
I think these 5 lines could be written as:

```ruby
Post.where(world_readable: true).where('slug = ? OR id = ?', params[:id], params[:id]).first!
```

`first!` raises the `RecordNotFound` and doesn't return `nil` (like `first` does).